### PR TITLE
Add auto completion for commands that can select pods

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
For better user experience.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by ymesika — APPROVED** [review]
LGTM

File: releasenotes/notes/48603.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+releaseNotes:
+- |
+  **Added** completion for Kubernetes pods for some commands that can select pods, such as `istioctl proxy-config <pod>`.`
```
Comment (GregHanson) [inline releasenotes/notes/48603.yaml:6]: typo?  I don't see the `completion` library changes in `istioctl/pkg/proxyconfig/proxyconfig.go`, did you mean `istioctl proxy-status`?

Comment (hanxiaop) [reply releasenotes/notes/48603.yaml:6]: Yes this is a typo, thanks!

**Review by GregHanson — APPROVED** [review]
lgtm

**Pull request merged**